grails-app/controllers/org/pih/warehouse/api/LocationApiController.groovy

```diff
@@ -103,7 +103,12 @@ class LocationApiController extends BaseDomainApiController {
 
         bindLocationData(existingLocation, jsonObject)
 
-        if (!existingLocation.validate() || !existingLocation.save(failOnError: true)) {
+        if (existingLocation.validate() && !existingLocation.hasErrors()) {
+            if (existingLocation?.address?.validate() && !existingLocation?.address?.hasErrors()) {
+                existingLocation.address.save()
+            }
+            existingLocation.save()
```

Comment:
Need to handle the case when address fails validation.

Reply:
Also we should check whether the location validation handles both location and any associations.